content.js

```diff
@@ -29,34 +47,58 @@ function toggleFiles() {
 }
 
 function addToggleBtn() {
-	const toggleBtn = createHtml('<a class="hide-files-btn btn btn-sm">Show dotfiles</a>');
+	const toggleBtn = createHtml(`<a class="hide-files-btn btn btn-sm">${label()}</a>`);
 	const btnContainer = document.querySelector('.file-navigation .right');
 
 	if (document.querySelector('.hide-files-btn')) {
+		addToggleBtnEvents();
 		return;
 	}
 
 	if (btnContainer) {
 		// insert after
 		btnContainer.insertBefore(toggleBtn, btnContainer.children[0]);
+		addToggleBtnEvents();
+	}
+}
 
-		const btn = document.querySelector('.hide-files-btn');
+function addToggleBtnEvents() {
+	const btn = document.querySelector('.hide-files-btn');
 
+	if (btn) {
 		btn.addEventListener('click', () => {
-			isHidden = !isHidden;
-			btn.textContent = isHidden ? 'Show dotfiles' : 'Hide dotfiles';
+			toggleOn = !toggleOn;
+			btn.textContent = label();
 			toggleFiles();
 		});
 	}
 }
 
+function label() {
+	return toggleOn ? 'Show dotfiles' : 'Hide dotfiles';
+}
+
 function trigger() {
 	addToggleBtn();
 	toggleFiles();
 }
 
 document.addEventListener('DOMContentLoaded', () => {
-	trigger();
-
 	new MutationObserver(trigger).observe(document.querySelector('#js-repo-pjax-container'), {childList: true});
+
+	window.chrome.storage.sync.get({
+		visibility: 'hidden',
+		ignoreRegEx: ''
```

Comment:
Do you think we should add a small "Restore Defaults" button somewhere hiding at the bottom? People aren't likely to remember the default RegExp we provided, and if they mess up with their RegExp (which is super easy to do), then they have lost all ability to restore the defaults. The thing about the chrome.storage defaults is that they only work on the initial load of the extension, so as long as the extension is still installed, they will never get that default value we supplied back.